src/inmanta/data/__init__.py

```diff
@@ -1835,6 +1827,42 @@ def make_hash(self):
         resources = await cls.select_query(query, values)
         return resources
 
+    @classmethod
+    async def get_resources_in_latest_version(
+        cls,
+        environment: uuid.UUID,
+        resource_type: Optional[m.ResourceType] = None,
+        attributes: Dict[PrimitiveTypes, PrimitiveTypes] = {},
+    ) -> List["Resource"]:
+        """
+        Returns the resources in the latest version of the configuration model of the given environment, that satisfy these
+        constraints.
+
+        :param environment: The resources should belong to this environment.
+        :param resource_type: The environment should have this resource_type.
+        :param attributes: The resource should contain these attribute-value pairs. Only
+        """
+        values = [cls._get_value(environment)]
+        query = f"""
+            SELECT *
+            FROM {Resource.table_name()} AS r1
+            WHERE r1.environment=$1 AND r1.model=(SELECT MAX(r2.model)
+                                                  FROM {Resource.table_name()} AS r2
+                                                  WHERE r2.environment=$1)
+        """
+        if resource_type:
+            query += " AND r1.resource_type=$2"
+            values.append(cls._get_value(resource_type))
+
+        result = []
+        async with cls._connection_pool.acquire() as con:
+            async with con.transaction():
+                async for record in con.cursor(query, *values):
+                    resource = cls(from_postgres=True, **record)
+                    if util.is_sub_dict(attributes, resource.attributes):
```

Comment:
+1: filtering in memory to prevent database injection, nice.

Given the expected size of the result set: good solution.

Perhaps add a comment to make this very explicit